### DISCUSSION TITLE
Is there a way to have a tokio future that signals if a socket enters error state (EPOLLERR)?

### DISCUSSION THREAD

**Discussion by liormal** [tokio-rs/tokio#4572] [Q&A]
I have a TcpStream that uses TCP_KEEPALIVE.
I only receive the connection closed error when I attempt to write to the socket.
There is a way to get signals about error events without trying to read/write.

**Comment by Darksonn** [comment]
There's already some discussion on this here: #3467. One thing you might try is to call `TcpStream::ready` and inspect the returned `Ready` value.
